index.d.ts

```diff
@@ -0,0 +1,373 @@
+
+  /**
+   * The line number of the location
+   */
+  line: number;
+
+  /**
+   * The 0-indexed column of the location
+   */
+  column: number;
+}
+
+export interface SourceSpan {
+  /**
+   * The text between `start` and `end` locations.
+   * This is from the exact line and column
+   */
+  text: string;
+
+  /**
+   * The start location of the `@warn` or `@debug` message
+   */
+  start: SourceLocation;
+
+  /**
+   * The end location of the `@warn` or `@debug` message
+   */
+  end: SourceLocation;
+
+  /**
+   * The location of the file with the message
+   */
+  url: string;
+
+  /**
+   * The text from `start` to `end`. However, unlike `text`, it outputs each full line
+   */
+  context: string;
+}
+
+export interface Result {
+  /**
+   * The compiled CSS.
+   *
+   * Write this to a file, or serve it out as needed.
+   */
+  css: Buffer;
+
+  /**
+   * The source map.
+   */
+  map?: Buffer;
+  stats: {
+    /**
+     * The path to the scss file, or `data` if the source was not a file.
+     */
+    entry: string;
+
+    /**
+     * `Date.now()` before the compilation.
+     */
+    start: number;
+
+    /**
+     * `Date.now()` after the compilation.
+     */
+    end: number;
+
+    /**
+     * `end - start`
+     */
+    duration: number;
+
+    /**
+     * Absolute paths to all related files in no particular order.
+     */
+    includedFiles: string[];
+  };
+}
+
+export function render(
+  options: Options,
+  callback: (exception: SassException, result: Result) => void
+): void;
+export function renderSync(options: Options): Result;
+
+export namespace types {
+  abstract class SassType {}
+
+  interface Null extends SassType {
+    NULL: Null;
+  }
+
+  const Null: Null;
+
+  class Number implements SassType {
+    constructor(value: number, unit?: string);
+    getValue(): number;
+    setValue(value: number): void;
+    getUnit(): string;
+    setUnit(unit: string): void;
+  }
+
+  class String implements SassType {
+    constructor(value: string);
+    getValue(): string;
+    setValue(value: string): void;
+  }
+
+  class Boolean<T extends boolean = boolean> implements SassType {
+    constructor(value: T);
+    getValue(): T;
+    static readonly TRUE: Boolean<true>;
+    static readonly FALSE: Boolean<false>;
+  }
+
+  class HslColor implements SassType {
+    constructor(h: number, s: number, l: number, a?: number);
+    getH(): number;
+    setH(value: number): void;
+    getS(): number;
+    setS(value: number): void;
+    getL(): number;
+    setL(value: number): void;
+    getA(): number;
+    setA(value: number): void;
+  }
+
+  class RgbColor implements SassType {
+    constructor(r: number, g: number, b: number, a?: number);
+    getR(): number;
+    setR(value: number): void;
+    getG(): number;
+    setG(value: number): void;
+    getB(): number;
+    setB(value: number): void;
+    getA(): number;
+    setA(value: number): void;
+  }
+
+  class List<T extends SassType = SassType> implements SassType {
```

Comment:
Sass doesn't consider its lists or maps to be typed, so I don't think this API should either.

Reply:
So they should just be `List<unknown>` and `Map<unknown>`, is that right?